batch/batch/cloud/gcp/driver/create_instance.py

```diff
@@ -264,9 +264,9 @@ def scheduling() -> dict:
 iptables --table nat --append POSTROUTING --source 172.20.0.0/15 --jump MASQUERADE
 
 # [public]
-# Block public traffic to the metadata server
-iptables --append FORWARD --source 172.21.0.0/16 --destination 169.254.169.254 --jump DROP
-# But allow the internal gateway
+# Send public jobs' metadata server requests to the batch worker itself
+iptables --table nat --append PREROUTING --source 172.21.0.0/16 --destination 169.254.169.254 -p tcp -j REDIRECT --to-ports 5555
```

Comment:
What part of this command instructs iptables to direct this request at the worker's IP rather than 169.254.169.254?

Reply:
This was an oversight because I started listening on all IPs. I will try to redirect this to localhost and listen only on loopback.

Reply:
Leaving a comment here for posterity:

The `REDIRECT` target changes the destination IP address of the packet to the IP address of the incoming interface. In this case, the destination IP address would be set to the IP address assigned to the host-side of the job's network namespace's veth pair. The worker specifically binds to this address before the job starts to provide the metadata server API. This the worker must specifically bind to an IP address per active job that requires a metadata server, but this seems to be a natural use case for `aiohttp` and it's coroutine-based architecture, as there should be no more overhead than what it takes to bind to a socket. This has the additional benefit that our firewall rules already ensure that no initiating packets from other jobs or the internet can reach that IP address.